solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/inbound/BatchCollector.java

```diff
@@ -55,26 +57,39 @@ public void addToBatch(MessageContainer messageContainer) {
 	 * @return true if available (batch may be empty).
 	 */
 	public boolean isBatchAvailable() {
+		// must be able to return an empty batch so that polled consumers don't block indefinitely
 		return isBatchAvailableInternal() && (!pruneStaleMessages() || isBatchAvailableInternal());
 	}
 
-	public boolean isBatchAvailableInternal() {
+	boolean isBatchAvailableInternal() {
 		if (batchedMessages.size() < consumerProperties.getBatchMaxSize()) {
-			long batchTimeDiff = System.currentTimeMillis() - timeSentLastBatch;
-			if (consumerProperties.getBatchTimeout() == 0 || batchTimeDiff < consumerProperties.getBatchTimeout()) {
-				if (logger.isTraceEnabled()) {
-					logger.trace(String.format("Collecting batch... Size: %s, Time since last batch: %s ms",
-							batchedMessages.size(), batchTimeDiff));
+			switch (consumerProperties.getBatchWaitStrategy()) {
+				case RESPECT_TIMEOUT -> {
+					long batchTimeDiff = System.currentTimeMillis() - batchCollectionStartTimestamp;
+					if (consumerProperties.getBatchTimeout() == 0 || batchTimeDiff < consumerProperties.getBatchTimeout()) {
+						if (!isLastMessageNull) {
+							LOGGER.trace("Collecting batch... Size: {}, Time elapsed: {} ms",
+									batchedMessages.size(), batchTimeDiff);
+						}
+						return false;
+					} else {
+						LOGGER.trace(
+								"Batch timeout reached <time elapsed: {} ms>, processing batch of {} messages...",
+								batchTimeDiff, batchedMessages.size());
+					}
+				}
+				case IMMEDIATE -> {
+					if (isLastMessageNull) {
+						LOGGER.trace("Last message was null, processing batch of {} messages...", batchedMessages.size());
```

Comment:
What if it'd say: ```No more messages are available on the endpoint, processing batch of {} messages..."```

The above links with the doc and I feel like 'last message was null' might not be obvious to users.

Reply:
Updated the message